aws-api/src/main/java/com/amplifyframework/api/aws/DomainType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The type of domain specified in API endpoint.
+ */
+public enum DomainType {
+    /**
+     * Invalid or unsupported domain type.
+     */
+    INVALID,
+
+    /**
+     * Standard domain type composed of randomly generated 26 char subdomain.
+     */
+    STANDARD,
+
+    /**
+     * Custom domain defined by user. It must be "https" and have path "graphql" in URL.
+     */
+    CUSTOM;
+
+    private static final String STANDARD_ENDPOINT_REGEX =
+            "^https:\\/\\/\\w{26}\\.appsync\\-api\\.\\w{2}(?:(?:\\-\\w{2,})+)\\-\\d\\.amazonaws.com\\/graphql$";
+
+    /**
+     * Get Domain type based on defined endpoint.
+     * @param endpoint Endpoint defined in config.
+     * @return {@link DomainType} based on supplied endpoint.
+     */
+    static DomainType from(String endpoint) {
+        if (!(endpoint.startsWith("https://") && endpoint.endsWith("/graphql"))) {
+            return INVALID;
+        }
```

Comment:
iOS isn't doing any schema or path checks, this seems like safe validation to add though. should other platforms be doing this? and would either of these two validation checks be changed in the future? What if the endpoint comes in as `wss://` already? what if the path is ever changed to "/graphql2" by AppSync?

Reply:
These validations has been confirmed with the appsync. I agree with you argument that the path might be changed and even though that would mean the requirements will be changed, having this validation removed will simply mean that it might fail in other locations.
Endpoints should not be sent as `wss://` to this method as per the definition and scope of this particular method. If there is any such usecase, this method can be called by replacing it with `https://` before passing as a parameter.